modules/launcher/src/main/java/org/mule/runtime/module/launcher/MuleContainer.java

```diff
@@ -217,6 +230,52 @@ protected void init(String[] args) throws IllegalArgumentException, Initialisati
     artifactResourcesRegistry.getMemoryManagementService().initialise();
     artifactResourcesRegistry.inject(artifactResourcesRegistry.getContainerProfilingService());
     initialiseIfNeeded(artifactResourcesRegistry.getContainerProfilingService());
+
+    if (isFipsEnabled()) {
+      configureSecurityManager();
+      setSecurityAlgorithm();
+    }
+  }
+
+  private void configureSecurityManager() throws InitialisationException {
+    try {
+      Class<?> classDef =
+          Class.forName(System.getProperty(PROPERTY_FIPS_PROVIDER, "org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider"));
+      Constructor<?> constructor = classDef.getConstructor();
+      Provider fipsProvider = (Provider) constructor.newInstance();
+      java.security.Security.insertProviderAt(fipsProvider, 1);
+
+      classDef =
+          Class.forName(System.getProperty(PROPERTY_JSSE_PROVIDER, "org.bouncycastle.jsse.provider.BouncyCastleJsseProvider"));
+      constructor = classDef.getConstructor();
+      Provider jsseProvider = (Provider) constructor.newInstance();
+      jsseProvider.setProperty("fips", "BCFIPS");
+      java.security.Security.insertProviderAt(jsseProvider, 2);
+
+      Provider sunJsseProvider = Security.getProvider("SunJSSE");
```

Comment:
This one should be:
```suggestion
      Provider sunJsseProvider = Security.getProvider(SUN_JSSE_PROVIDER);
```
right?